src/dds/rs-dds-auto-calibration.cpp

```diff
@@ -0,0 +1,97 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+
+#include "rs-dds-auto-calibration.h"
+#include <rsutils/string/from.h>
+#include <rsutils/json.h>
+#include <src/ds/d500/d500-auto-calibration.h>
+
+
+namespace librealsense
+{
+dds_auto_calibrated::dds_auto_calibrated() :
+    _auto_calib_capability(std::make_shared<d500_auto_calibrated>()) {}
+
+
+std::vector<uint8_t> dds_auto_calibrated::run_on_chip_calibration(int timeout_ms, std::string json, 
+    float* const health, rs2_update_progress_callback_sptr progress_callback)
+{
+    return _auto_calib_capability->run_on_chip_calibration(timeout_ms, json, health, progress_callback);
+}
+
+
+std::vector<uint8_t> dds_auto_calibrated::run_tare_calibration(int timeout_ms, float ground_truth_mm, std::string json, float* const health, rs2_update_progress_callback_sptr progress_callback)
+{
+    throw not_implemented_exception(rsutils::string::from() << "Tare Calibration not applicable for this device");
```

Comment:
As a proxy class, why not leave it to the implementation to throw if it does not implement?

Reply:
done